e2e-pw/src/oss/specs/regression-tests/grid-page.spec.ts

```diff
@@ -0,0 +1,77 @@
+import { test as base } from "src/oss/fixtures";
+import { GridPom } from "src/oss/poms/grid";
+import { ModalPom } from "src/oss/poms/modal";
+import { getUniqueDatasetNameWithPrefix } from "src/oss/utils";
+
+const test = base.extend<{ grid: GridPom; modal: ModalPom }>({
+  grid: async ({ page }, use) => {
+    await use(new GridPom(page));
+  },
+  modal: async ({ page }, use) => {
+    await use(new ModalPom(page));
+  },
+});
+
+const datasetName = getUniqueDatasetNameWithPrefix("grid-page");
+const groupDatasetName = getUniqueDatasetNameWithPrefix("grid-page-group");
+
+test.beforeAll(async ({ fiftyoneLoader }) => {
+  await fiftyoneLoader.executePythonCode(`
+    import fiftyone as fo
+
+    dataset = fo.Dataset("${datasetName}")
+    dataset.persistent = True
+    dataset.add_samples([fo.Sample(filepath=f"{i}.png", group=1, i=i) for i in range(0, 21)])
+
+    view = dataset.group_by("group", order_by="i")
+    dataset.save_view("group", view)
+
+    group_dataset = fo.Dataset("${groupDatasetName}")
+    group_dataset.persistent = True
+    group_dataset.add_group_field("group")
+    group_samples = []
+    group = fo.Group()
+    for i in range(0, 21):
+        group_samples.append(fo.Sample(filepath=f"{i}.png", group=group.element(str(i))))
+
+    group_dataset.add_samples(group_samples)
+  `);
+});
+
+test("grid has correct second page (all 21 samples)", async ({
+  fiftyoneLoader,
+  grid,
+  page,
+}) => {
+  await fiftyoneLoader.waitUntilLoad(page, datasetName);
```

Comment:
`WaitUntilGridVisibleOptions` is the new `waitUntilLoad` I think. You might have to rebase develop to get the new name